Firestore/Example/Tests/Local/FSTRemoteDocumentCacheTests.mm

```diff
@@ -50,15 +57,41 @@
 static const char *kLongDocPath = "a/b/c/d/e/f";
 static const int kVersion = 42;
 
+void ExpectMapHasDocs(XCTestCase *self,
+                      const MaybeDocumentMap &map,
+                      const std::vector<Document> &expected,
+                      bool exactly) {
+  if (exactly) {
+    XCTAssertEqual(map.size(), expected.size());
+  }
+  for (const Document &doc : expected) {
+    absl::optional<MaybeDocument> actual = map.get(doc.key());
```

Comment:
Perhaps check that `actual` has a value before dereferencing below? It looks like a failing test could cause a crash (here and below).